### PR TITLE
Cdhj 101 feature/view users

### PR DESCRIPTION
**Pull request by MajdKhodr**

### CODE REVIEW

File: app/Controllers/Http/UserController.js [hunk 1]
```diff
@@ -77,6 +125,26 @@ class UserController {
...
+	 *
+	 * @param {Object} Context The context object.
+	 */
+	async edit ({ params, view, auth }) {
+		// Retrieves user object
+		const user = await User.findBy('id', params.id);
+		var layoutType = '';
+
+		// Check type of user in order to render the correct layout
+		if (auth.user.role === 1) {
```
Comment (dynamic11) [inline app/Controllers/Http/UserController.js]: You should check if the user is editing his own account

File: app/Controllers/Http/UserController.js [hunk 2]
```diff
@@ -230,12 +298,13 @@ class UserController {
...
 		const user = await User.find(Number(params.id));
 		var canEdit = 0;
+		var isAdmin = false;
 		var layoutType = '';
 		// check if admin is viewing their own profile
 		if (auth.user.role === 1) {
 			layoutType = 'layouts/adminLayout';
 			canEdit = 1;
-
+			isAdmin = true;
```
Comment (dynamic11) [inline app/Controllers/Http/UserController.js]: isAdmin is redundant. canEdit should be giving the same logic. A perso should be able to see their own bookings so use canEdit as your boolean

File: app/Controllers/Http/UserController.js (hunk 1 above)
Comment (MajdKhodr) [reply app/Controllers/Http/UserController.js]: isn't that already checked when the button is rendered, with the canEdit variable?

Comment (dynamic11) [reply app/Controllers/Http/UserController.js]: that blocks the button from rendering but it doesn't plot someone from entering the page by just typing in the url

Comment (dynamic11) [reply app/Controllers/Http/UserController.js]: That blocks the button from renedering but it doesnt block the user from entering by typing in the url

**Pull request merged**